pkg/devfile/adapters/kubernetes/component/kaniko_build.go

```diff
@@ -0,0 +1,314 @@
+package component
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/openshift/odo/pkg/devfile/adapters/common"
+	"github.com/openshift/odo/pkg/devfile/adapters/kubernetes/utils"
+	"github.com/openshift/odo/pkg/kclient"
+	"github.com/openshift/odo/pkg/log"
+	"github.com/openshift/odo/pkg/sync"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/klog"
+)
+
+const (
+	kanikoSecret          = "kaniko-secret"
+	buildContext          = "build-context"
+	buildContextMountPath = "/root/build-context"
+	kanikoSecretMountPath = "/root/.docker"
+	completionFile        = "/tmp/complete"
+	builderServiceAccount = "builder"
+	containerName         = "build"
+	initContainerName     = "init"
+)
+
+var (
+	buildContextVolumeMount = corev1.VolumeMount{Name: buildContext, MountPath: buildContextMountPath}
+	kanikoSecretVolumeMount = corev1.VolumeMount{Name: kanikoSecret, MountPath: kanikoSecretMountPath}
+
+	secretGroupVersionResource = schema.GroupVersionResource{Group: "", Version: "v1", Resource: "secrets"}
+	defaultId                  = int64(0)
+)
+
+func (a Adapter) runKaniko(parameters common.BuildParameters, isImageRegistryInternal bool) error {
+	if isImageRegistryInternal {
+		if err := a.createDockerCfgSecretForInternalRegistry(parameters.EnvSpecificInfo.GetNamespace()); err != nil {
+			return errors.Wrap(err, "failed to create dockerconfig secret")
+		}
+	}
+
+	labels := map[string]string{
+		"component": a.ComponentName,
+	}
+
+	if err := a.createKanikoBuilderPod(labels, initContainer(initContainerName), builderContainer(containerName, parameters.Tag, isImageRegistryInternal), regcredName); err != nil {
+		return errors.Wrap(err, "error while creating kaniko builder pod")
+	}
+
+	podSelector := fmt.Sprintf("component=%s", a.ComponentName)
```

Comment:
we have different kind of selectors present in our component/labels package and similarly in application/labels. please consider using those